src/main/java/com/store/book/service/impl/BookServiceImpl.java

```diff
@@ -0,0 +1,49 @@
+package com.store.book.service.impl;
+
+import com.store.book.dto.BookDto;
+import com.store.book.dto.CreateBookRequestDto;
+import com.store.book.exception.EntityNotFoundException;
+import com.store.book.mapper.BookMapper;
+import com.store.book.model.Book;
+import com.store.book.repository.BookRepository;
+import com.store.book.service.BookService;
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
+
+    @Override
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        book.setIsbn("" + (long) (Math.random() * 9_000_000_000_000L) + 1_000_000_000_000L);
```

Comment:
It seems logic of setting Isbn will be more appropriate in mapper to obtain ready to use model